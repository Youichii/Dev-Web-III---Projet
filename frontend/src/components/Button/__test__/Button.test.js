import React from 'react';
import ReactDOM from 'react-dom';
import Button from '../Button';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer  from "react-test-renderer";

//npm test to start all tests
afterEach(cleanup);
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button></Button>, div)
})

it("render button text correctly", () => {
    const {getByTestId} = render(<Button color="green" text='Hello' onClick={()=> {console.log("hello")}} ></Button>);
    expect(getByTestId('button')).toHaveTextContent("Hello")
})

it("render button color correctly", () => {
    const {getByTestId} = render(<Button color="green" text='Hello' ></Button>);
    expect(getByTestId('button')).toHaveStyle("backgroundColor : green")
})

it("matches snapshot 1", () => {
    const tree = renderer.create(<Button text="save" color="blue"></Button>).toJSON(); //convert to a virtualDOM object
    expect(tree).toMatchSnapshot(); //create folder __snapshot__
}) // to update a Snapshot if it failed press -u in the terminal

it("matches snapshot 2", () => {
    const tree = renderer.create(<Button text="Hello" color="blue"></Button>).toJSON(); //convert to a virtualDOM object
    expect(tree).toMatchSnapshot();
}) // creating a 2nd snapshot

