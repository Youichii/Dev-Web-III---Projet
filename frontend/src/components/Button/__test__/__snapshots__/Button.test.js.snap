// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<button
  className="btn"
  data-testid="button"
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  save
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="btn"
  data-testid="button"
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  Hello
</button>
`;
